mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp

```diff
@@ -678,6 +678,14 @@ spirv::Deserializer::getConstant(uint32_t id) {
   return constIt->getSecond();
 }
 
+std::optional<std::pair<uint32_t, Type>>
+spirv::Deserializer::getConstantCompositeReplicate(uint32_t id) {
+  auto constIt = constantCompositeReplicateMap.find(id);
+  if (constIt == constantCompositeReplicateMap.end())
+    return std::nullopt;
+  return constIt->getSecond();
```

Comment:
nit: I'd define `it` in the if condition itself and flip the check:
```suggestion
  if (auto it = constantCompositeReplicateMap.find(id); it != constantCompositeReplicateMap.end())
    return it->second;
  return std::nullopt;
```